src/components/tl/mlx5/mcast/tl_mlx5_mcast_helper.h

```diff
@@ -0,0 +1,316 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5.h"
+
+ucc_status_t ucc_tl_probe_ip_over_ib(const char* ib_dev_list, struct
+                                     sockaddr_storage *addr);
+
+#ifndef MCAST_H_
+#define MCAST_H_
+#include "tl_mlx5_mcast_progress.h"
+#include "utils/ucc_math.h"
+
+#define MIN(a,b) DO_OP_MIN(a,b)
+#define MAX(a,b) DO_OP_MAX(a,b)
+
+static inline int ucc_tl_mlx5_mcast_poll_send(ucc_tl_mlx5_mcast_coll_comm_t *comm)
+{
+    struct ibv_wc wc;
+    int           num_comp;
+    
+    num_comp = ibv_poll_cq(comm->scq, 1, &wc);
```

Comment:
it is going to be 1 in the future as well.

Reply:
why not using `POLL_PACKED` or another value ?

Reply:
This helps us to keep the stream of mcast packet live. As soon as we see that pending_send is reduced, we have the green light to place another mcast packet on the wire. ucc_tl_mlx5_mcast_poll_send() is a polling function and it is the choice of the user of this function to keep calling it as many times as it needs to clear the pending mcast sends.

Reply:
I am not sure to understand why batching the completion is problematic. In my opinion it just potentially reduces the numbers of PCI transactions, while achieving the same goal. If you call `ibv_poll_cq(comm->scq, 8, &wc);` but there is only 1 completion in the cq, then you'll get only this completion and num_comp will be set to one; If there are more than 1, then you'll get all of them (up to 8) at once. Why isn't it desirable?

Anyway, I don't want to insist on this because it might really not be so important, but I wanted to understand if there was a particular reason for keeping 1 here.

Reply:
_I wanted to understand if there was a particular reason for keeping 1 here._
Yes, I agree, the CPU is not the bottleneck here, it really does not make much difference to say use 1 or more in this polling function.  In the future, we may want to increase the value and see if it has a better performance impact. The only reason for picking value 1 is to make this function as light as possible and let the user decide how many times to call it. Using value 1 lets us avoid adding a loop in this polling function. It can also minimize the latency of ibv_poll_cq().